File: helpers/get_spins.py
import numpy as np
# def one_component(Psi_B, Psi_C):
# 	return np.imag(
# 		np.conj(Psi_B) * Psi_C
# 		- Psi_B * np.conj(Psi_C)
# 	)
def get_spins(Psi3):
	# return np.stack([
	# 	one_component(Psi[(j+1) % 3], Psi[(j+2)%3]) for j in axes
	# ], axis=0)
	return np.real(np.cross(1j * Psi3, np.conj(Psi3), axis=0))

File: helpers/getSpins.m
function Spins = getSpins(Psi, simConfig, normalized)
%spinRaw - Compute a 3x1 cells array of double arrays containing the spin in each component.
%
% Syntax: Spins = spinRaw(Psi)
%
% Long description
	
Spins = cell(1, 3);
for j = 1:3
	PsiB = Psi{rem(j, 3) + 1};
	PsiC = Psi{rem(j + 1, 3) + 1};
	if normalized
		normConst = 1.;
	else
		normConst = simConfig.dx^3;
	end
	Spins{j} = imag(conj(PsiB) .* PsiC - PsiB .* conj(PsiC)) * normConst;
end
end